extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts

```diff
@@ -177,4 +192,50 @@ export class HostedPluginRemote {
         return [].concat.apply([], [...this.pluginsMetadata.values()]);
     }
 
+    /**
+     * Sends request to retreive plugin resource from its sidecar.
+     * Returns undefined if plugin doesn't run in sidecar or doesn't exist.
+     * @param pluginId id of the plugin for which resource should be retreived
+     * @param resourcePath relative path of the requested resource based on plugin root directory
+     */
+    public requestPluginResource(pluginId: string, resourcePath: string): Promise<Buffer | undefined> {
+        if (this.hasEndpoint(pluginId) && resourcePath) {
+            return new Promise<Buffer | undefined>((resolve, reject) => {
+                const endpoint = this.hostedPluginMapping.getPluginsEndPoints().get(pluginId);
+                const targetWebsocket = this.endpointsSockets.get(endpoint);
+                if (!targetWebsocket) {
+                    reject(new Error(`No websocket connection for plugin: ${pluginId}`));
+                }
+
+                this.resourceRequests.set(this.getResourceRequestId(pluginId, resourcePath), resolve);
+                targetWebsocket.send(JSON.stringify({
+                    'internal': {
+                        'method': 'getResource',
+                        'pluginId': pluginId,
+                        'path': resourcePath
+                    }
+                }));
+            });
+        }
+        return undefined;
+    }
+
+    /**
+     * Handles all responses from all remote plugins.
+     * Resolves promise from getResource method with requested data.
+     */
+    onGetResourceResponse(pluginId: string, resourcePath: string, resource: Buffer | undefined): void {
+        const key = this.getResourceRequestId(pluginId, resourcePath);
```

Comment:
I would use concatenation or something like
`${pluginId}_${resourcePath}`
But it's more question of habits.